### PR TITLE
feat: Allow for using '{namespace}' and '{type}' placeholders for 'ordId' and 'partOfPackage'

### PR DESCRIPTION
**Pull request by mlakov**
# Allow `{namespace}` Placeholder in `ordId` and `partOfPackage` Fields

### New Feature

✨ Introduced support for the `{namespace}` (and `{type}`) placeholder in `ordId` and `partOfPackage` fields across all ORD resource templates. After resource creation, placeholders are resolved by substituting them with the actual `appConfig.ordNamespace` and resource type values. This eliminates the need to hardcode namespace values in templates and configurations.

A dedicated `lib/common/placeholders.js` utility module handles all placeholder resolution using a simple `replace(value, context)` function.

### Changes

* `lib/common/placeholders.js`: **New file** — implements a generic `replace(value, context)` function that substitutes `{key}` placeholders in a string using the provided context object.
* `lib/templates/api-resource.js`: Applies `placeholders.replace()` to `ordId` and `partOfPackage` after resource construction. Simplified `_getPackageID` by removing the early return for missing `packageIds`. Spreads `ordExtensions` while omitting the `ordId` key to prevent double-application.
* `lib/templates/entity-type.js`: Updated `RESOLVERS.ordId` and `RESOLVERS.partOfPackage` to resolve placeholders via the new utility. Added `appConfig` parameter to `RESOLVERS.ordId` and `RESOLVERS.version`. Spreads ORD extensions while omitting already-resolved resolver keys.
* `lib/templates/event-resource.js`: Updated `RESOLVERS.ordId` and `RESOLVERS.partOfPackage` to use placeholder resolution. Added `@ORD.Extensions.exposedEntityTypes` and `@ORD.Extensions.partOfGroups` override support. Propagates `appConfig` to `exposedEntityTypes` resolver.
* `lib/templates/integration-dependency.js`: Updated `RESOLVERS.ordId` and `RESOLVERS.partOfPackage` to resolve placeholders. Added support for `appConfig.env.integrationDependency.aspects` override.
* `lib/templates/package.js`: Applies `placeholders.replace()` to `ordId` after package creation. Replaced `productOrdId` (single value) with `products` (array) to support multiple products in `partOfProducts`.
* `lib/templates/product.js`: Applies `placeholders.replace()` to `ordId` after product creation.
* `__tests__/unit/common/placeholders.test.js`: **New file** — comprehensive unit tests for the `placeholders.replace` utility covering single/multiple replacements, unknown placeholders, edge cases (`null`, `undefined`, empty string).
* `__tests__/unit/templates/api-resource.test.js`: Added tests for placeholder replacement in `@ORD.Extensions.ordId` and `partOfPackage`. Removed test case for `_getPackageID` returning `undefined` when no `packageIds` are provided.
* `__tests__/unit/templates/entity-type.test.js`: Added `appConfig` parameter to all `RESOLVERS` calls. Added tests for `@ORD.Extensions.ordId` and `partOfPackage` placeholder replacement.
* `__tests__/unit/templates/event-resource.test.js`: Added tests for `{namespace}:{type}` placeholder resolution in `ordId` and `partOfPackage`.
* `__tests__/unit/templates/integration-dependency.test.js`: Added tests for placeholder resolution in `ordId` and `partOfPackage` overrides.
* `__tests__/unit/templates/package.test.js`: Updated test to use renamed `products` array parameter. Added test for placeholder resolution in package `ordId` overrides.
* `__tests__/unit/templates/product.test.js`: Added snapshot test for `{namespace}:{type}` placeholder resolution in product `ordId`.
* `__tests__/bookshop/srv/admin-service.cds`: Added example usage of `{namespace}:{type}` placeholder in `@ORD.Extensions.ordId`.

- [ ] 🔄 Regenerate and Update Summary

<details>
<summary>PR Bot Information</summary>

**Version:** `1.21.4`

- Event Trigger: `issue_comment.edited`
- Correlation ID: `79545727-bce8-437d-a092-fd6bf8011d69`
</details>

### CODE REVIEW

File: lib/templates/api-resource.js [hunk 1]
```diff
@@ -115,9 +118,9 @@ const createAPIResourceTemplate = (srvDefinition, appConfig) => {
         const { apiProtocol, entryPoints, hasResourceDefinitions } = protocolResult;
         const protocolExtensions = readORDExtensions(srvDefinition, `@protocol('${apiProtocol}').ORD.Extensions.`);
         const ordId =
-            protocolExtensions.ordId ||
-            ordExtensions.ordId ||
-            `${appConfig.ordNamespace}:apiResource:${cleanServiceName}${index === 0 ? "" : `-${apiProtocol}`}:${version}`;
+            protocolExtensions.ordId?.replace(/\{type}/g, "apiResource")?.replace(/\{namespace}/g, namespace) ??
+            ordExtensions.ordId?.replace(/\{type}/g, "apiResource")?.replace(/\{namespace}/g, namespace) ??
+            `${namespace}:apiResource:${cleanServiceName}${index === 0 ? "" : `-${apiProtocol}`}:${version}`;
```
Comment (swennemers) [inline lib/templates/api-resource.js]: I would like some extraction of this replacement logic into util module, or similar.

**Review by swennemers — DISMISSED** [review]

**Review by zongqichen — DISMISSED** [review]
LGTM

**Review by zongqichen — APPROVED** [review]
LGTM
